### PR TITLE
move some battle data to c

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

**Review by ethteck — CHANGES_REQUESTED** [review]
LGTM aside from all newly named symbols should be in symbol_addrs

**Comment by nanaian** [comment]
can we automate adding to symbol_addrs 👉 👈

File: src/code_16c8e0.c [hunk 1]
```diff
@@ -1,4 +1,137 @@
...
+#include "script_api/battle.h"
+
+f32 D_802809F0 = 0.0f;
+s8 D_802809F4 = 0;
+s8 D_802809F5 = 0;
+s16 D_802809F6 = -1;
+s16 D_802809F8 = 0;
+u16 D_802809FA = 0;
+
+s32 D_802809FC[] = {
```
Comment (pixel-stuck) [inline src/code_16c8e0.c:12]: probably label these pointers?

File: src/code_16c8e0.c [hunk 2]
```diff
@@ -1,4 +1,137 @@
...
+    if (SI_VAR(1) == 0) {
+        JumpToGoal(256, 20, 0, 0, 1);
+    } else {
+        JumpToGoal(256, 20, 0, 0, 1);
+    }
+    GetActorPos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+    ForceHomePos(256, SI_VAR(0), SI_VAR(1), SI_VAR(2));
+});
+
+u8 D_80280CE0[] = { 0, 0, 0, 0 };
```
Comment (pixel-stuck) [inline src/code_16c8e0.c:60]: file padding? Also, why u8?

File: src/code_190B20.c [hunk 3]
```diff
@@ -1,5 +1,454 @@
...
+s16 D_802835D8[] = { 0, -12 };
+
+s32 D_802835DC[] = {
+    0x00000000, 0x40900000, 0x00000000, 0x3F800000, 0x40800000, 0x00000000, 0x40000000, 0x40400000, 0x00000000, 0x40400000, 0x40000000, 0x00000000, 0x40600000, 0x3F800000, 0x00000000, 0x40800000, 0x00000000, 0x00000000, 0x40900000, 0x00000000, 0x00000000, 0x40A00000, 0x00000000, 0x00000000, 0x40900000, 0x00000000, 0x00000000, 0x40800000, 0x00000000, 0x00000000, 0x40600000, 0xBF800000, 0x00000000, 0x40400000, 0xC0000000, 0x00000000, 0x40000000, 0xC0400000, 0x00000000, 0x3F800000, 0xC0800000, 0x00000000, 0x00000000, 0xC0900000, 0x00000000,
+};
+
+s32 D_80283690[] = {
+    0x3F800000, 0x3F800000, 0x3F800000, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F666666, 0x3F666666, 0x3F666666, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F800000, 0x3F800000, 0x3F800000, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F666666, 0x3F666666, 0x3F666666, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F800000, 0x3F800000, 0x3F800000, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F666666, 0x3F666666, 0x3F666666, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F8CCCCD, 0x3F800000, 0x3F800000, 0x3F800000, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F4CCCCD, 0x3F666666, 0x3F666666, 0x3F666666,
```
Comment (pixel-stuck) [inline src/code_190B20.c:435]: these are definitely floats

File: src/code_190B20.c [hunk 4]
```diff
@@ -1,5 +1,454 @@
...
+// TODO: split 0x8028358C
+s32 D_80283524[] = {
+    0x00050000, 0xFFEC002D, 0x0000FFFB, 0x00550000, 0x000A007D, 0x00000019, 0x000A0032, 0xFFEC0032, 0x002DFFFB, 0x005A0032, 0x000A0082, 0x00370019, 0x000F0055, 0xFFEC0037, 0x0050FFFB, 0x005F0055, 0x000A0087, 0x005A0019, 0x000F007D, 0xFFEC0037, 0x0078FFFB, 0x005F007D, 0x000A0087, 0x00820019, 0x00690000, 0x00000000, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 0x8028358C, 0x8028358C, 0x8028358C, 0x8028358C, 0x8028358C, 0x8028358C, 0x8028358C, 0x8028358C, 0x00000000,
+};
+
+s16 D_802835D0[] = { 0x1C, 0x28 };
+s16 D_802835D4[] = { 0, -2 };
+s16 D_802835D8[] = { 0, -12 };
+
+s32 D_802835DC[] = {
```
Comment (pixel-stuck) [inline src/code_190B20.c:430]: these too

File: src/code_190B20.c [hunk 5]
```diff
@@ -1,5 +1,454 @@
...
+        .posOffset = { 0, 0, 0 },
+        .targetOffset = { 0, 32 },
+        .opacity = 0xFF,
+        .idleAnimations = bMarioIdleAnims,
+        .defenseTable = bMarioDefenseTable,
+    },
+};
+
+// TODO: split 0x8028358C
+s32 D_80283524[] = {
```
Comment (pixel-stuck) [inline src/code_190B20.c:422]: hmm, we should at least add the pointers in here to undefined_syms.txt, unless there's some reason not to?

File: src/code_190B20.c [hunk 6]
```diff
@@ -1,5 +1,454 @@
...
+    .spinSmashReq = 4,
+    .powerBounceChance = 80,
+
+    .size = { 33, 43 },
+    .hpBarOffset = { 0, 0 },
+    .statusIconOffset = { -10, 30 },
+    .statusMessageOffset = { 10, 30 },
+};
+
+ActorPartDesc bMarioParts[] = {
```
Comment (pixel-stuck) [inline src/code_190B20.c:410]: what does the `b` prefix mean?

**Review by pixel-stuck — CHANGES_REQUESTED** [review]

File: src/code_190B20.c (hunk 6 above)
Comment (nanaian) [reply src/code_190B20.c:410]: Battle

File: src/code_16c8e0.c (hunk 2 above)
Comment (nanaian) [reply src/code_16c8e0.c:60]: No,
```
glabel add_cam_zoom
/* 17CE94 8024E5B4 3C028028 */  lui       $v0, %hi(D_80280CE0)
/* 17CE98 8024E5B8 80420CE0 */  lb        $v0, %lo(D_80280CE0)($v0)
```

Comment (nanaian) [reply src/code_16c8e0.c:60]: The latter 3 bytes are possibly file padding, but I'd rather wait for us to be certain

Comment (ethteck) [reply src/code_16c8e0.c:60]: agr33d

**Pull request merged**